src/public/modules/forms/base/directives/submit-form.directive.js

```diff
@@ -94,9 +92,7 @@ function submitFormDirective(
 
       scope.formLogin = function (authType, rememberMe) {
         if (authType === 'MyInfo') {
-          return MyInfoRedirect({
-            formId: scope.form._id,
-          })
+          return createRedirectURL(scope.form._id)
```

Comment:
Does angularjs need us to wrap this promise in `$q.when` since angularjs doesn't have support for native promises?

Reply:
just noted that our application has support for promises due to polyfills, but $q will still trigger a digest cycle, correct me if i'm wrong; so it's still needed?